decentraland/unity-explorer
# fix: avatar step sounds being unpredictable when in-between states

## What does this PR change?

+ Added a new int to the avatar animator to represent the desired movement type (walk, jog, run or none), as before we were using the blend value, but it was not predictable and specially when decelerating, did not represent the current movement type.
+ Added also a small time control for steps, so steps cant happen in too short amounts of time, same for jumps and landings.

In the small refactor branch this is cleaned up and extracted properly.
Fixes this issue -> https://github.com/decentraland/unity-explorer/issues/849

## How to test the changes?

1- Have perfect hearing
2- Intercalate between running, jogging, walking and whatnot, to try to hear simultaneous steps happening. Now we shouldn't when before we could.
The audio heard will always correspond to the movement type indicated by the input so, for example, it wont matter that we are accelerating from walking to running, if we are pressing shift, we will hear running sound from the beginning and vice versa.
This should reduce greatly the ghost sounds heard when in-between states
